crates/base_db/src/fixture.rs

```diff
@@ -175,19 +175,39 @@ impl ChangeFixture {
 
             if let Some(krate) = meta.krate {
                 let crate_name = CrateName::normalize_dashes(&krate);
-                let crate_id = crate_graph.add_crate_root(
-                    file_id,
-                    meta.edition,
-                    Some(crate_name.clone().into()),
-                    meta.cfg,
-                    meta.env,
-                    Default::default(),
-                );
-                let prev = crates.insert(crate_name.clone(), crate_id);
-                assert!(prev.is_none());
-                for dep in meta.deps {
-                    let dep = CrateName::normalize_dashes(&dep);
-                    crate_deps.push((crate_name.clone(), dep))
+                match crates.entry(crate_name.clone()) {
+                    Entry::Occupied(_) => {
+                        if !meta.deps.is_empty() {
+                            assert_eq!(
+                                crate_deps,
+                                meta.deps
+                                    .iter()
+                                    .map(|dep| (
+                                        crate_name.clone(),
+                                        CrateName::normalize_dashes(dep)
+                                    ))
+                                    .collect::<Vec<_>>(),
+                                "Crate {} has two modules with different dependencies in metadata",
+                                krate,
+                            )
+                        }
+                    }
+                    Entry::Vacant(v) => {
+                        let new_crate_id = crate_graph.add_crate_root(
+                            file_id,
+                            meta.edition,
+                            Some(crate_name.clone().into()),
+                            meta.cfg,
+                            meta.env,
+                            Default::default(),
+                        );
+                        v.insert(new_crate_id);
+
+                        for dep in meta.deps {
+                            let dep = CrateName::normalize_dashes(&dep);
+                            crate_deps.push((crate_name.clone(), dep))
+                        }
+                    }
```

Comment:
Hm, why do we need to change anything here at all? *Just* the following diff seem to work for me:

```diff
diff --git a/crates/ide_assists/src/handlers/auto_import.rs b/crates/ide_assists/src/handlers/auto_import.rs
index 5ccd7f7a2..e04711067 100644
--- a/crates/ide_assists/src/handlers/auto_import.rs
+++ b/crates/ide_assists/src/handlers/auto_import.rs
@@ -918,6 +918,9 @@ fn main() {
             auto_import,
             r"
 //- /lib.rs crate:dep
+pub mod formatters;
+
+//- /formatters.rs
 pub struct FMT;
 pub struct fmt;
 
@@ -926,7 +929,7 @@ fn main() {
     FMT$0;
 }
 ",
-            r"use dep::FMT;
+            r"use dep::formatters::FMT;
 
 fn main() {
     FMT;
```

I suspect  that we just support multifie crates in fixtures without any any extra actions required?

Reply:
Indeed, so I've completely misunderstood the `//-` fixture format.
Thanks for stopping me 🙂

Reply:
might make sense to clarify the docs then?

Reply:
Yes, and reread the updated version finally.